scripts/standby_node_health_check/standby_node_health_check.py

```diff
@@ -0,0 +1,450 @@
+        return interfaces
+
+    @classmethod
+    def _parse_ip_link_output(cls, ip_link_output: str):
+        interfaces = {}
+        try:
+            link_data = json.loads(ip_link_output)
+            for entry in link_data:
+                ifname = entry.get("ifname")
+                operstate = entry.get("operstate", "").lower()  # Get state in lowercase
+                link_type = entry.get("link_type", "").lower()
+
+                if ifname and link_type == "ether":
+                    interfaces[ifname] = operstate
+
+        except json.JSONDecodeError:
+            print("Error while parssing ib link show command")
+        return interfaces
+
+    def _check_eth_interfaces(self):
+        interfaces_status = self._run_and_parse_ip_link_show()
+        result = True
+        for interface in self._mgmt_interfaces:
+            if interface not in interfaces_status:
+                print(f"{interface} is not in the list of management interfaces")
+                result = False
+            elif interfaces_status[interface] != "up":
+                print(
+                    f"Interface {interface} is not active {interfaces_status[interface]}"
+                )
+                result = False
+        if result:
+            print("All given ether interfaces are active")
+        return result
+
+    @classmethod
+    def _check_if_ha_is_enabled(cls):
+        command = cls.UFM_HA_CLUSTER_COMMAND.format("is-ha")
+        ret_code, stdout = cls._run_command(command)
+        if ret_code != 0 or stdout.lower() != "yes":
+            print("HA is not enabled")
+            return False
+        print("HA is enabled")
+        return True
+
+    def run_basic_checks(self):
+        result = True
+        checks = [
+            self._check_ib_interfaces,
+            self._check_eth_interfaces,
+            self._check_if_ha_is_enabled,
+        ]
+        for check in checks:
+            if not check():
+                result = False
+        return result
+
+    @classmethod
+    def check_if_standby_node(cls):
+        command = cls.UFM_HA_CLUSTER_COMMAND.format("is-master")
+        _, stdout = cls._run_command(command)
+        # Not checking the ret code since it is 1 when running on the standby node
+        if stdout != "standby":
+            print(f"Not a standby node but {stdout}")
+            return False
+        print("On standby node")
+        return True
+
+    @classmethod
+    def _check_pcs_status(cls):
+        command = "pcs status"
+        return_code, _ = cls._run_command(command)
+        if return_code != 0:
+            print(f"pcs status is not ok, return code is {return_code}")
+            return False
+        print("PCS status is OK")
+        return True
+
+    @classmethod
+    def _check_corosync_rings_status(cls):
+        ret_code, corosync_output = cls._run_command("corosync-cfgtool -s")
+        if ret_code != 0:
+            print("Failed to run corosync-cfgtool -s")
```

Comment:
Please output to the user more general messages related to HA interfaces rather than corosync/RING/...
You may write to log more informative messages for out debug.

Reply:
Fixed